components/AlertsDashboard.vue

```diff
@@ -68,6 +68,101 @@ const showSlider = ref(false);
 const route = useRoute();
 const router = useRouter();
 
+const isMapeo = ref(false);
+
+/**
+ * Selects and zooms to an alert feature based on its ID
+ */
+const selectInitialAlertFeature = (alertId: string) => {
+  const allFeatures = [
+    ...props.alertsData.mostRecentAlerts.features,
+    ...props.alertsData.previousAlerts.features,
+  ];
+  const feature = allFeatures.find((f) => f.properties?.alertID === alertId);
+  if (feature && feature.properties) {
+    // Find the appropriate layer ID for this feature by checking both recent and previous layers
+    const geometryType = feature.geometry.type.toLowerCase();
+    const recentLayerId = `most-recent-alerts-${geometryType}`;
+    const previousLayerId = `previous-alerts-${geometryType}`;
+
+    // Check if feature exists in recent layer
+    const isInRecentLayer = props.alertsData.mostRecentAlerts.features.some(
+      (f) => f.properties?.alertID === feature.properties?.alertID,
+    );
+
+    // Select feature in the correct layer
+    const layerId = isInRecentLayer ? recentLayerId : previousLayerId;
+    selectFeature(feature, layerId);
+
+    // Zoom to the feature
+    if (feature.geometry.type === "Point") {
+      const [lng, lat] = feature.geometry.coordinates;
+      map.value.flyTo({ center: [lng, lat], zoom: 13 });
+    } else if (
+      feature.geometry.type === "Polygon" ||
+      feature.geometry.type === "MultiPolygon"
+    ) {
+      const bounds = bbox(feature);
+      map.value.fitBounds(bounds, { padding: 50 });
+    } else if (feature.geometry.type === "LineString") {
+      const [lng, lat] = calculateLineStringCentroid(
+        feature.geometry.coordinates,
+      );
+      map.value.flyTo({ center: [lng, lat], zoom: 13 });
+    }
+    isMapeo.value = false;
+  }
+};
+
+/**
+ * Selects and zooms to a Mapeo feature based on its document ID
+ */
+const selectInitialMapeoFeature = (mapeoDocId: string) => {
+  const mapeoFeature = props.mapeoData?.find((f) => f.ID === mapeoDocId);
+
+  if (mapeoFeature) {
+    const geometryType = mapeoFeature.geotype as
```

Comment:
Add a code comment explaining to the reader why you need to create a new `feature`, like why you can't just use `mapeoFeature` directly.


As I understand, there may be more than 1 reason, mostly the 3 reasons presented here: https://github.com/ConservationMetrics/gc-explorer/pull/109#issuecomment-2985123992 - we might link to that comment as well